pelican/tests/test_readers.py

```diff
@@ -795,6 +803,24 @@ def test_typogrify_dashes_config(self):
         self.assertEqual(page.content, expected)
         self.assertEqual(page.title, expected_title)
 
+    def test_metadata_has_no_discarded_data(self):
+        md_filename = 'article_with_markdown_and_empty_tags.md'
+        page = self.read_file(
+            path=md_filename,
+            cache_name='cache',
+            CACHE_CONTENT=True,
+            LOAD_CONTENT_CACHE=True)
+
+        file_path = _path(md_filename)
+        cached_metadata = self._reader._cache[file_path][1][1]
```

Comment:
Better to use the public interface ([`get_cached_data`](https://github.com/getpelican/pelican/blob/f862d64b7a492f2af24e1228790a1d75371a1112/pelican/readers.py#L571)) instead of accessing the internal cache implementation.

Reply:
Yep, that makes sense.